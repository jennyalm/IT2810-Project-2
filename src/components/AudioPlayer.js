import React from 'react'

class AudioPlayer extends React.Component {
    constructor(props){
        super(props)
        this.state = {}
        this.getSound = this.getSound.bind(this)

    }

    // pauses the audio and then loads it. This is needed to rerender the sound if it changes. It's not enough to just edit the src in source.
    getSound(){
        this.playAudio.pause();
        this.playAudio.load();
    }

    // runs getSound() whenever the component updates it's props and current soundCategory not equals the previous one.
    componentDidUpdate(prevProps){
        if(this.props.soundCategory !== prevProps.soundCategory ) {
            this.getSound()
        }
    }

    // runs getSound() when the component is made.
    componentDidMount(){
        this.getSound()
    }

    // returns a div with a audio tag and a source. The source gets the correct path from public with soundCategory and soundName
    render() {
        return (
            <div style={{width: "100%", height: "100%"}}>
                <audio
                    controls
                    ref={(playAudio) => { this.playAudio = playAudio; }}
                    style={{width: "100%", height: "100%"}}
                >
                    <source ref="audio"
                            src={process.env.PUBLIC_URL + "/Sound/" + this.props.soundCategory + "/" + this.props.soundName}
                            type="audio/mpeg"
                    />
                </audio>
            </div>

        );
    }
}

export default AudioPlayer;