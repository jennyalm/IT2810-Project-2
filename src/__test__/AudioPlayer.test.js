import React from 'react'
import renderer from 'react-test-renderer'

import AudioPlayer from '../components/AudioPlayer';

test('AudioPlayer snapshot test', () => {



    const tree = renderer.create( <AudioPlayer/> ).toJSON();

    expect(tree).toMatchSnapshot();
});