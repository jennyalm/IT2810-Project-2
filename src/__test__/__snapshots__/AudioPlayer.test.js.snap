// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AudioPlayer snapshot test 1`] = `
<div>
  <audio
    controls={true}
  >
    <source
      src="./Sound/undefined/undefined"
      type="audio/mpeg"
    />
  </audio>
</div>
`;
